### PR TITLE
X-Basic Auth Extension

### PR DESCRIPTION
**Pull request by jashieh**
Creates x-basic auth extension, which replaces the `Authorization` header with the value from `x-auth` when triggered in the authentication chain.

### CODE REVIEW

**Comment by vwangg2** [comment]
changes look good and approved but we should merge this in to druid-infra as discussed in standup

**Review by vwangg2 — CHANGES_REQUESTED** [review]
changing to request changes (see druid-infra comment)

File: source/DruidXbasicExtension/pom.xml [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.amazon.solutions.druid.extensions</groupId>
+    <artifactId>druid-xbasic</artifactId>
+    <name>druid-xbasic</name>
+    <description>druid-xbasic</description>
+    <version>25.0.0</version>
```
Comment (msalman-atl) [inline source/DruidXbasicExtension/pom.xml:29]: This is not a core extension and should have versioning of its own.

File: source/DruidXbasicExtension/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+`{
+    "druid.auth.authenticator.xbasic.type": "xbasic",
+    "druid.auth.authenticator.xbasic.authorizerName": "xbasic",
+    "druid.auth.authorizer.xbasic.type": "xbasic"
+}`
+- Update the `authenicatorChain` in `druidCommonRuntimeConfig` to have `xbasic` before `basic` auth:
+`{
+  "druid.auth.authenticatorChain": ["xbasic", "basic"]
+}`
+- When making a request to druid, set the header `x-auth` to your basic auth token. The extension will replace the
```
Comment (msalman-atl) [inline source/DruidXbasicExtension/README.md:21]: Can you add an example here using curl or slauth curl.

**Review by msalman-atl — CHANGES_REQUESTED** [review]
We should move this to internal repo.

**Pull request closed without merging**